src/wallet/api/wallet.cpp

```diff
@@ -690,6 +690,20 @@ bool WalletImpl::store(const std::string &path)
     return m_status == Status_Ok;
 }
 
+bool WalletImpl::change_password(const std::string& new_password)
+{
+    try
+    {
+        m_wallet->rewrite(m_wallet->path(), new_password);
+        m_wallet->store();
+    } catch (const std::exception &e) {
+        LOG_ERROR("Error while changing password: " << e.what());
+        m_status = Status_Error;
+        m_errorString = e.what();
+    }
+    return m_status == Status_Ok;
```

Comment:
m_status is not set on the sucess path.

Reply:
In fact, that might be on purpose to not lose a previous failure ? If that is the case, I'd return at start I think.

Reply:
Oops! Fixed

Reply:
Hmm, now I' confused which convention is the right one, e.g. `setPassword` vs `daemonBlockChainHeight`.

Reply:
Hmm. I just looked and I don't get the difference between setPassword and change_password. The new one doesn't save the password, which seems odd. Can you explain the intent for the new function and differences ?